lib/galaxy/webapps/galaxy/api/dataset_collections.py

```diff
@@ -64,6 +69,44 @@ def create(self, trans: ProvidesHistoryContext, payload: dict, **kwd):
         return dictify_dataset_collection_instance(dataset_collection_instance,
                                                    security=trans.security, parent=create_params["parent"])
 
+    @expose_api
+    def update(self, trans: ProvidesHistoryContext, payload: dict, id):
+        """
+        Iterate over all datasets of a collection and copy datasets with new attributes to a new collection.
+        e.g attributes = {'dbkey': 'dm3', 'file_ext' = 'txt'}
+
+        * PUT /api/dataset_collections/{hdca_id}:
+            create a new dataset collection instance.
+        """
+
+        # TODO: make sure attributes are valid
```

Comment:
That seems like a pretty important TODO. What happens if you try to update the id for instance ?